sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -163,6 +163,48 @@ class CollationStringExpressionsSuite
     })
   }
 
+  test("Support Lpad string expressions with collation") {
+    // Supported collations
+    case class PadTestCase[R](s: String, len: Int, pad: String, c: String, result: R)
+    val testCases = Seq(
+      PadTestCase("", 5, " ", "UTF8_BINARY", "     "),
+      PadTestCase("abc", 5, " ", "UNICODE", "  abc"),
+      PadTestCase("abc", 5, "XY", "UTF8_BINARY_LCASE", "XYabc"),
+      PadTestCase("abc", 5, "123", "UNICODE_CI", "12abc"),
+      PadTestCase("abc", 2, " ", "UTF8_BINARY", "ab"),
+      PadTestCase("abc", 2, "XY", "UNICODE", "ab"),
+      PadTestCase("abc", 2, "123", "UTF8_BINARY_LCASE", "ab"),
+      PadTestCase("abc", 2, "123", "UNICODE_CI", "ab")
```

Comment:
since we're already using this compact subset of testcases (which is great), let's spice these up a bit

mix in some case & accent variation, as well as variable byte-length characters here (but no need to add more test cases)

Reply:
Done! @uros-db